### DISCUSSION TITLE
mini.files ignore files in .gitignore

### DISCUSSION THREAD

**Discussion by justEstif** [nvim-mini/mini.nvim#396] [Q&A] [answered]
Hello, 

I have switched from nvim-tree to mini.files, and it has been generally seamless.  

I have found the [toggle dotfiles](https://github.com/echasnovski/mini.nvim/blob/a4241b6f51393d5e4d91a12d94e386bae287dae5/doc/mini-files.txt#L366-L387) function in the docs and it has been helpful. 

Would it be possible for the filter to depend on .gitignore similar to nvim-tree?

**Comment by echasnovski** [comment, marked as answer]
> Hello,

Hi!

> Would it be possible for the filter to depend on .gitignore similar to nvim-tree?

Sure, this is possible with custom `content.filter` function. Albeit it is slightly tricky as `content.filter` takes single file system entry as input.

To respect '.gitignore', its code should find proper '.gitignore' file(s) and decide whether input path should be ignored.

Alternatively, it could use [`git check-ignore`](https://git-scm.com/docs/git-check-ignore) shell command for each file path.

Another approach is to hard-code some common block patterns for yourself.

------

That said, it won't be built-in neither into 'mini.files' code nor documentation, as I strongly believe that file explorer should show **all** files present on the disk.
